### PR TITLE
Bug 1741681: Bump cadvisor [4.1.z]

### LINKED ISSUE #23492: Missing network metrics from cadvisor after some time
Hi,

I'm using okd 3.11 with cri-o. After some uptime of origin-node, cadvisor doesn't report all network metrics (e.g. `container_network_transmit_packets_total`).

##### Version
I've build own RPMs for Origin and installed this RPM yesterday (25.07.2019) without any changes from branch `release-3.11`. My hope was a possible fix.
```
11:01 $ oc version
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://s-openshift.mycompany.com:443
openshift v3.11.0+3b2d3b6-227
kubernetes v1.11.0+d4cacc0
```

##### Steps To Reproduce
1. RSH to Prometheus instance
2. `curl -s -k -H "Authorization: Bearer [My-Token]" https://10.20.15.42:10250/metrics/cadvisor | grep container_network_transmit_packets_total | grep namespace=\"thanos\"`
```
container_network_transmit_packets_total{container_name="POD",id="/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod6588346a_aefa_11e9_ab2c_005056945e02.slice/crio-45848b690fe11481394b28c108b1fb25f31c5aaa41ff93ad4f56c71501dc90b6.scope",image="",interface="eth0",name="k8s_POD_thanos-querier-6d98c99694-dj8hg_thanos_6588346a-aefa-11e9-ab2c-005056945e02_0",namespace="thanos",pod_name="thanos-querier-6d98c99694-dj8hg"} 334942
container_network_transmit_packets_total{container_name="POD",id="/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod69a3dfe5_aefa_11e9_ab2c_005056945e02.slice/crio-d07f32d58bd0b4240685147aa1af32e36f7558684c600a397c99af68a16c40c8.scope",image="",interface="eth0",name="k8s_POD_minio-2_thanos_69a3dfe5-aefa-11e9-ab2c-005056945e02_0",namespace="thanos",pod_name="minio-2"} 2.115004e+06
```
3. wait some time e.g. 5 hours.
4. Again without output `curl -s -k -H "Authorization: Bearer [My-Token]" https://10.20.15.42:10250/metrics/cadvisor | grep container_network_transmit_packets_total | grep namespace=\"thanos\"`

##### Current Result
After some hours network metrics disappear for interface eth0
Network_eth0

##### Expected Result
Stable network metrics from cadvisor.

### PR DESCRIPTION
**Pull request by sjenning**
master PR https://github.com/openshift/origin/pull/23585

pull in https://github.com/openshift/google-cadvisor/pull/7

The delta is larger because a `make update-deps` on a clean `origin/master` makes a lot of changes and the `github.com/apcera/gssapi` dep needed to be redirected to the openshift fork because that repo doesn't exist anymore.  I've split up the bump(*) on the clean master into a separate commit.

Fixes #23492

@derekwaynecarr @deads2k @mfojtik

### CODE REVIEW

**Comment by sjenning** [comment]
@deads2k note the removal of `DefaultServingCertificate` in the `openshift/api` bump.  Want to make sure that is ok.

**Comment by sjenning** [comment]
well, I guess it isn't fine
```
 ++ Building go targets for linux/amd64: cmd/hypershift vendor/k8s.io/kubernetes/cmd/hyperkube
# github.com/openshift/origin/pkg/admission/customresourcevalidation/apiserver
pkg/admission/customresourcevalidation/apiserver/validate_apiserver.go:120:26: spec.ServingCerts.DefaultServingCertificate undefined (type "github.com/openshift/origin/vendor/github.com/openshift/api/config/v1".APIServerServingCerts has no field or method DefaultServingCertificate)
pkg/admission/customresourcevalidation/apiserver/validate_apiserver.go:131:29: newSpec.ServingCerts.DefaultServingCertificate undefined (type "github.com/openshift/origin/vendor/github.com/openshift/api/config/v1".APIServerServingCerts has no field or method DefaultServingCertificate) 
```
need to adjust the pin for `openshift/api` for 4.1

**Comment by sjenning** [comment]
https://github.com/openshift/api/pull/375 should not have been picked back to openshift/api release 4.1 branch...

**Comment by eparis** [comment]
This need engineering review/lgtm/approval
/retest

File: pkg/admission/customresourcevalidation/apiserver/validate_apiserver.go [hunk 1]
```diff
@@ -41,7 +41,6 @@ func (a apiserverV1) ValidateCreate(uncastObj runtime.Object) field.ErrorList {
 	}
 
 	errs = append(errs, validation.ValidateObjectMeta(&obj.ObjectMeta, false, customresourcevalidation.RequireNameCluster, field.NewPath("metadata"))...)
-	errs = append(errs, validateAPIServerSpecCreate(obj.Spec)...)
```
Comment (mfojtik) [inline pkg/admission/customresourcevalidation/apiserver/validate_apiserver.go]: @sjenning how is this related to the cadvisor bump?

File: glide.lock [hunk 2]
```diff
@@ -2040,7 +2041,7 @@ imports:
   subpackages:
   - config/v1beta1
 - name: k8s.io/kubernetes
-  version: cc2afe3181bd5593ad5c03e9a768e3ff270ec343
+  version: fd7008ebbbb179b36367d02fc929ca7eb514d0b0
```
Comment (mfojtik) [inline glide.lock]: hmm why this moved and what is this level?

Comment (mfojtik) [reply glide.lock]: /cc @deads2k @sttts 

Is the bot broken?

Comment (sttts) [reply glide.lock]:
```
$ git log cc2afe3181bd5593ad5c03e9a768e3ff270ec343..fd7008ebbbb179b36367d02fc929ca7eb514d0b0 --oneline
fd7008ebbbb Merge pull request #23517 from sttts/sttts-procmount-4.1
cf981128196 Merge pull request #23524 from rphillips/backport_4.1/80657
dbd6628aba9 UPSTREAM: 80657: add UID to kubelet_container_log_filesystem_used_bytes metric
d78ddd3c037 UPSTREAM: 78882: Fix incorrect procMount defaulting
f27555ff6d6 Merge pull request #23367 from RobertKrawitz/upstream-79391-4.1
4597d2ea87f bump
32ecfe871e5 UPSTREAM: 79391: Don't use strategic merge patch on Node.Status.Addresses
```

File: pkg/admission/customresourcevalidation/apiserver/validate_apiserver.go (hunk 1 above)
Comment (sjenning) [reply pkg/admission/customresourcevalidation/apiserver/validate_apiserver.go]: it isn't, but https://github.com/openshift/api/pull/375 went into `openshift/api` branch `release-4.1` and after the bump, the build failed without this change

**Pull request merged**